packages/template-retail-react-app/app/page-designer/assets/image-with-text/index.jsx

```diff
@@ -6,8 +6,16 @@
  */
 import React from 'react'
 import PropTypes from 'prop-types'
-import {Box, Image, Text} from '@chakra-ui/react'
+import {Box, Image, Link as ChakraLink, Text} from '@chakra-ui/react'
 import Link from '../../../components/link'
+import {isAbsoluteURL} from '../utils'
+
+const LinkWrapper = ({ITCLink, children}) => {
+    const chakraLinkWrapper = (children) => <ChakraLink href={ITCLink}>{children}</ChakraLink>
+    const linkWrapper = (children) => <Link to={ITCLink}>{children}</Link>
+
+    return isAbsoluteURL(ITCLink) ? chakraLinkWrapper(children) : linkWrapper(children)
+}
```

Comment:
If you look at the Link component from the template components folder you'll see that that component has similar logic built into it that you are duplicating here maybe? 🤔  

I think there is a prop called useNavLink that you might be able to take advantage of.

Reply:
I don't think that is the case.

Reading our custom link component uses a react-router Link or NavLink component (used for the Breadcrumbs menu) with the Chakra Link styles.
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/3396472bf97e5b62f658d0d74e5e2ea6995d64d7/packages/template-retail-react-app/app/components/link/index.jsx#L18-L19

There's no logic to assert if the `url` string is absolute, and if so, use a Chakra UI Link component instead of react-router component.

Reply:
The prop `useNavLink` only determines if we use react-router's Link vs NavLink component.

Reply:
I've resolved a similar issue recently by using the `as` prop. Since the `...props` spread is after the `as={useNavLink ? ...}` prop (L21 in the snippet below), you can override it by passing a custom `as` to the `Link` component.

https://github.com/SalesforceCommerceCloud/pwa-kit/blob/6b9d4369b1460023de233e8b60b31feda38ebf55/packages/template-retail-react-app/app/components/link/index.jsx#L13-L26

Meaning you could possibly adjust your current logic to something like:

```tsx
import Link from '../path/to/Link'

const linkProps = ({ITCLink}) => {
    return isAbsoluteURL(ITCLink) ? { as: ChakraLink } : {}
}

() => <Link {...linkProps(/* args */ )} >...</Link>
```

Additionally, I added a `LinkComponent?: ComponentWithAs<'a'>` prop to our `<Link/>`, which has a default value of `<ChakraLink />`. This allows us to override the UI of the link (e.g. using Chakra UI's `<LinkOverlay />` component) while retaining the react-router link logic from the `as` prop.

Reply:
Thanks @breadadams for the feedback. I played with the idea and you're right, we can override our custom react-router `<Link />` component logic with the `ChakraLink` logic by using `as={ChakraLink}`. The declared `as` prop passed to the `<Link />` component will override the `as` prop with `...props`  spread.
But then we need to modify the `<Link />` to conditionally use `href` or `to` props depending on if we use `ChakraLink` or a react-router-like `Link` component.

The `LinkComponent` idea to override styles and keep the react-router Link logic makes sense as well, but today we use the `layerStyle` prop pattern to override styles.

The scope of the PR is to fix the Link in the `ImageWithText` Page designer component. We plan to extract the Page designer code from the template in the future. In this context, I think it's better to solve the problem in the `ImageWithText` component file. But both ideas are good suggestions to make our custom `<Link />` component more flexible in future extensibility work.